scheduler/src/state_manager/mod.rs

```diff
@@ -218,6 +218,50 @@ impl StateManager {
                     ))
                     .await;
             }
+
+            let deleting_instances: Vec<&mut WorkloadInstance> = workload
+                .instances
+                .iter_mut()
+                .filter_map(|(_, instance)| match instance.is_destroying() {
+                    true => Some(instance),
+                    false => None,
+                })
+                .collect();
+
+            for instance in deleting_instances {
+                let worker = workers.next().unwrap();
+
+                instance.set_worker(Some(worker.clone()));
+                // For now we don't check whether the instance is properly deleted, we assume it is
+                // as if we keep the destroying state, it will loop here and spam riklet of events
+                instance.set_status(ResourceStatus::Terminated);
+
+                info!("Deleting instance {}", instance.id.clone());
+
+                let _ = self
```

Comment:
why make an empty assignation here ?

Reply:
AFAIK it was due to memory management between Rust and async. Without it, the async task would have been dropped. Might not be relevant anymore, for safety lets' keep it.